### ISSUE TITLE
work-for-factions.js: Possible bug in rep gain calculations

### ISSUE THREAD

**Issue by mikeymikey64** [alainbryden/bitburner-scripts#353]
work-for-factions.js switches to a less profitable type of work in terms of rep gains, yet it states that it detected it as a more profitable work.

Faction "Bachman & Associates" Highest Aug Req: 375,000, Current Favor (79.97/150.00) Req: 365,668
INFO: Detected that "security" gives more rep than previous work "hacking". Switching...
Doing 'security' work for "Bachman & Associates" until 375,000 rep. Currently at 107,893, earning 13.948 rep/sec. (ETA: 5h 19m 10s)

**Comment by alainbryden** [comment]
Sorry, what's the indication that this *isn't* working as intended?

**Comment by mikeymikey64** [comment]
Ahh sorry for the confusion. Unfortunately I no longer have the log for it.

The general problem was work-for-faction.js switching to security work despite it giving lower rep gains. Security gave 13.9 while Hacking gave 40 something. Initially, I thought it must have wanted the other stats but the above log showed that it had believed that security gave more rep, which wasn't true.

**Comment by mehax** [comment]
Yes I'm able to reproduce it, however this bug will either fix itself after 1 minute or restart the script
```
Faction "Fulcrum Secret Technologies" Highest Aug Req: 1,500,000, Current Favor (73.66/150.00) Req: 379,978
Doing 'security' work for "Fulcrum Secret Technologies" until 379,978 rep. Currently at 249,057, earning 198.326 rep/sec. (ETA: 11m)
Doing 'security' work for "Fulcrum Secret Technologies" until 379,978 rep. Currently at 262,688, earning 200.435 rep/sec. (ETA: 9m 45s)
INFO: Detected that "hacking" gives more rep than previous work "security". Switching...
Doing 'hacking' work for "Fulcrum Secret Technologies" until 379,978 rep. Currently at 276,437, earning 151.399 rep/sec. (ETA: 11m 24s)
INFO: Detected that "security" gives more rep than previous work "hacking". Switching...
Doing 'security' work for "Fulcrum Secret Technologies" until 379,978 rep. Currently at 287,141, earning 335.240 rep/sec. (ETA: 4m 37s)
```

**Comment by mikeymikey64** [comment]
ah, I see. I tried moving it back to hacking contract manually but it swapped back. It did get fixed after I restarted but I never tried waiting XD  Thanks for looking into it.

**Comment by alainbryden** [comment]
I'm gonna leave this open and see if the game has any new API's that make this less of a guessing game
